libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -152,13 +153,13 @@ void IRGenerationContext::addToInternalDispatch(FunctionDefinition const& _funct
 	solAssert(functionType, "");
 
 	YulArity arity = YulArity::fromType(*functionType);
-
-	if (m_internalDispatchMap.count(arity) != 0 && m_internalDispatchMap[arity].count(&_function) != 0)
-		// Note that m_internalDispatchMap[arity] is a set with a custom comparator, which looks at function IDs not definitions
-		solAssert(*m_internalDispatchMap[arity].find(&_function) == &_function, "Different definitions with the same function ID");
-
-	m_internalDispatchMap[arity].insert(&_function);
-	enqueueFunctionForCodeGeneration(_function);
+	auto it = ranges::find(m_internalDispatchMap[arity], &_function);
+	if (it != ranges::end(m_internalDispatchMap[arity]))
+		solAssert(*it == &_function, "Different definitions with the same function ID");
+	else {
+		m_internalDispatchMap[arity].push_back(&_function);
```

Comment:
Do we have a test against duplicates in the dispatch? If not, please add a case like this:
```solidity
contract C {
    function a() internal {}
    function f() public {
        function () ptr1 = a;
        function () ptr2 = a;
    }
}
```

The easiest way to check would probably be to remove the check against duplicates in `IRGenerationContext::addToInternalDispatch()`, run semantic tests and see if any of them fails.

Reply:
We do not as far as I could see. I added the test and ran it as you suggested, removing the check against duplicates, and it throws an error as expected:
```
Invalid IR generated:
Error: Duplicate case "1" defined.
```